maintenance/wikia/makeImagesConsistentAgain.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+require_once __DIR__ . '/../Maintenance.php';
+
+class MakeImagesConsistentAgain extends Maintenance {
+
+	public function __construct() {
+		parent::__construct();
+		$this->setBatchSize( 200 );
+		$this->addOption( 'dry-run', 'Only output statistics without queueing tasks' );
+		$this->addOption( 'from', 'Process uploads later than this timestamp', true, true );
+		$this->addOption( 'until', 'Process uploads earlier than this timestamp', true, true );
+	}
+
+	public function execute() {
+		$dryRun = $this->hasOption( 'dry-run' );
+
+		if ( $dryRun ) {
+			$this->output( "Dry-run mode, no tasks will be queued!\n" );
+		}
+
+		$i = 0;
+		$batches = [];
+		$count = 0;
+		$total = 0;
+
+		// prepare no more than mBatchSize file operations for each background task
+		foreach ( $this->getAllOperations() as $operation ) {
+			if ( ++$count > $this->mBatchSize ) {
+				$i++;
+			}
+
+			$batches[$i][] = $operation;
+			$total++;
+		}
+
+		$this->output( "Got $total operations to synchronize; going to queue $i background tasks.\n" );
+
+		if ( !$dryRun ) {
+			foreach ( $batches as $batch ) {
+				$task = \Wikia\Tasks\Tasks\ImageSyncTask::newLocalTask();
+				$task->call( 'synchronize', $batch );
+				$task->queue();
+			}
+		} else {
+			var_dump( $batches );
+		}
+	}
+
+	private function getAllOperations() {
+		$uploads = 0;
+		foreach ( $this->getNewUploads() as $operation ) {
```

Comment:
Is it possible to get rid of the `foreach`:
```
$newUploads = $this->getNewUploads();
$uploads = count( $newUploads );
yield $newUploads;
```

And the same applies to deletes below. It seems the `foreach`s are used just to increment the counters.

Reply:
`getNewUploads` will also return a `Generator`, so we'll have to iterate through it anyways and `yield` its values

Reply:
Ah, all clear.